### PR TITLE
Add radio button widget

### LINKED ISSUE #6763: Add Radio Button Widgets
### Describe the feature you would like to be added.

Pyvista currently has a checkbox button widget. I think it could be useful to add a **radio button** widget - where only one button in a group of buttons can be on.  Example use case: switching active scalar of a displayed mesh.

### Links to VTK Documentation, Examples, or Class Definitions.

I have a basic working version implemented in my forked repo [under feat/radio_button](https://github.com/nathmay/pyvista/blob/1f17c70fc4d0d5a0c25b4755a1364242c20737fc/pyvista/plotting/widgets.py#L2628). 

### Pseudocode or Screenshots

**Example application 1:** Switching background color
```python
import pyvista as pv

plotter = pv.Plotter()

def set_bg(color):
  def wrapped_callback():
    plotter.background_color = color
  return wrapped_callback

plotter.add_radio_button_widget(set_bg('white'), 'bgcolor', position=(10.0,200.0), value=True, title="White")
plotter.add_radio_button_widget(set_bg('lightblue'), 'bgcolor', position=(10.0,140.0), title="Light Blue")
plotter.add_radio_button_widget(set_bg('pink'), 'bgcolor', position=(10.0,80.0), title="Pink")
plotter.show()
```

**Example application 2:** Switching scalar of plotted mesh
```python
import pyvista as pv

sphere = pv.Sphere()
# Add example scalars for demonstration
sphere['X_Values'] = sphere.points[:,0]
sphere['Y_Values'] = sphere.points[:,1]
sphere['Z_Values'] = sphere.points[:,2]

plotter = pv.Plotter()
sphere.set_active_scalars('X_Values')
actor = plotter.add_mesh(sphere)

# Callback function for changing active scalar
def set_scalar(scalar):
  def wrapped_callback():
    global actor
    plotter.remove_actor(actor)
    sphere.set_active_scalars(scalar)
    actor = plotter.add_mesh(sphere)
  return wrapped_callback

# Add radio buttons
plotter.add_radio_button_widget(set_scalar('X_Values'), 'scalar', position=(10.0,200.0),
                                title="X Values", value=True)
plotter.add_radio_button_widget(set_scalar('Y_Values'), 'scalar', position=(10.0,140.0),
                                title="Y Values")
plotter.add_radio_button_widget(set_scalar('Z_Values'), 'scalar', position=(10.0,80.0),
                                title="Z Values")
plotter.show()
```
image

### PR DESCRIPTION
**Pull request by nathmay**
### Overview

Potentially resolves #6763.

### Details

- `add_radio_button_widget`  method added in `WidgetHelper` class at `pyvista/plotting/widgets.py`.
- Implements radio button groups, where only one button in a group can be active at a time. A callback is performed when the active button is switched.

### CODE REVIEW

File: pyvista/plotting/widgets.py [hunk 1]
```diff
@@ -2623,6 +2625,202 @@ def _the_callback(widget, _event):
...
+                for j in range(size):
+                    distance = np.sqrt((i - size / 2) ** 2 + (j - size / 2) ** 2)
+                    if distance < rad_inner:
+                        arr[i, j] = fg_color
+                    elif rad_inner <= distance <= rad_inner + smooth:
+                        blend = (distance - rad_inner) / smooth
+                        arr[i, j] = (1 - blend) * fg_color + blend * bg_color
+                    elif rad_outer - 2 * smooth <= distance <= rad_outer:
+                        blend = abs(distance - rad_outer + smooth) / smooth
+                        arr[i, j] = (1 - blend) * fg_color + blend * bg_color
```
Comment (tkoyama010) [inline pyvista/plotting/widgets.py:2763]:
```suggestion
            for i, j in product(range(size), range(size)):
                distance = np.sqrt((i - size / 2) ** 2 + (j - size / 2) ** 2)
                if distance < rad_inner:
                    arr[i, j] = fg_color
                elif rad_inner <= distance <= rad_inner + smooth:
                    blend = (distance - rad_inner) / smooth
                    arr[i, j] = (1 - blend) * fg_color + blend * bg_color
                elif rad_outer - 2 * smooth <= distance <= rad_outer:
                    blend = abs(distance - rad_outer + smooth) / smooth
                    arr[i, j] = (1 - blend) * fg_color + blend * bg_color
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

File: pyvista/plotting/widgets.py (hunk 1 above)
Comment (tkoyama010) [reply pyvista/plotting/widgets.py]: @nathmay You can [incorporate feedback in your pull request using GitHub feature](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request) :)

**Comment by tkoyama010** [comment]
If you need assistance resolving an error, please feel free to mention us.

File: pyvista/plotting/widgets.py (hunk 1 above)
Comment (nathmay) [reply pyvista/plotting/widgets.py]: @tkoyama010 Thank you! I realized this after already making additional commits. Can you tell I'm new at this? 😀

File: tests/plotting/test_widgets.py [hunk 2]
```diff
@@ -332,6 +332,82 @@ def test_widget_closed(uniform):
...
+    p.add_mesh(uniform)
+    b1 = p.add_radio_button_widget(
+        callback=func, radio_button_group='group', value=True, size=50, position=(10.0, 10.0)
+    )
+    b2 = p.add_radio_button_widget(
+        callback=func, radio_button_group='group', size=50, position=(10.0, 70.0)
+    )
+    p.show(auto_close=False)
+    # Click b1 and switch active radio button
+    p.iren._mouse_left_button_click(35, 95)
```
Comment (tkoyama010) [inline tests/plotting/test_widgets.py:396]: Great test! I have one question. Could you explain how you calculated the button click positions? I’d like to be able to review how this was calculated later on.

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

File: tests/plotting/test_widgets.py (hunk 2 above)
Comment (nathmay) [reply tests/plotting/test_widgets.py:396]: Good point! First, I just noticed that my comment `# Click b1 ...` is incorrect. The test is clicking 'b2', and checking if active button changes from b1 to b2.

I calculate the button click position as: (button_x + button_size/2, button_y + size/2). (button_x, button_y) is the position of the button passed during button creation, and is the bottom left corner of the button bounding box.

In this test, b2 is defined with `size=50, position=(10.0,70.0)`. So the button click position, or the center of the button, is at `(10 + 50/2, 70 + 50/2) = (35, 95)`.

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Pull request merged**